include/nudb/impl/create.ipp

```diff
@@ -88,19 +88,25 @@ create(
         ec = error::invalid_block_size;
         return;
     }
+    bool edf = false;
+    bool ekf = false;
+    bool elf = false;
     {
         File df(args...);
         File kf(args...);
         File lf(args...);
         df.create(file_mode::append, dat_path, ec);
```

Comment:
Reviewing this, I noticed that `posix_file::create` tries to open the file twice: Once to determine if it exists, and then once with `O_CREAT` to create it. See here: https://github.com/vinniefalco/NuDB/blob/master/include/nudb/impl/posix_file.ipp#L63-L76

From the man pages using `O_EXCL` will: 

> Ensure that this call creates the file: if this flag is specified in conjunction with O_CREAT, and pathname already exists, then open() will fail.

These are the exact semantics we want. So we can replace those lines with a single `open` call with `O_CREAT | O_EXCL` set: 

    fd_ = ::open(path.c_str(), result.first | O_CREAT | O_EXCL);
    if(fd_ == -1)
        return last_err(ec);

This will do the right thing. You can cherry-pick bc2438d16844064dbebc78db6147097835faebea if you want this, or I can do a separate PR.